### ISSUE TITLE
SVG Parsing seems to miss every last (or first?) segment of a continuous loop.

### ISSUE THREAD

**Issue by thantik** [openhardwarecoza/LaserWeb3#14]
Here's an example of what I'm seeing.  Looks like every last/first segment of my SVG is being dropped.  Off-by-one error?  Only reason you don't see it on the P or R is that the travel move crosses the missing segment.

![laserweb](https://cloud.githubusercontent.com/assets/920031/15985609/3c095e38-2fc1-11e6-9896-37687636274c.png)

**Comment by ghost** [comment]
Shapes.autoclose...  If we always close then single lines or open ended
lines also gets closed.    But for closed lines and objects you want it to
autoclose.  I. E.  No win win situation of changing the default.  Its
something we have to let users select per object.  Or per file...

I.e something i need to do as part of that tabbed layers cam rework i was
so hyper to start on before the big comms disaster.

Long story short,  pushing ahead with Comms.  Then back to this.

**Comment by cojarbi** [comment]
Is there anyway to cheat in the meantime? I need to cut squares but its open.

**Comment by ghost** [comment]
Use dxf.. Dxf is infinitely more suited format for lasers than svg

**Comment by ghost** [comment]
![autoclose configurable](https://cloud.githubusercontent.com/assets/7695323/15987199/b8eeec74-3020-11e6-88c1-cf958b1a2ef3.PNG)

Added to settings -> tools the option to select yes or no...

**Comment by funinthefalls** [comment]
Good addition

**Comment by cojarbi** [comment]
I cant get any code from SVG with the new commit. Can anyone else confirm? Tried enabling and disabling Auto close

**Comment by thantik** [comment]
No gcode here either @cojarbi - Additionally, the new option seems to do nothing as far as closing loops.  Changed setting, reloaded page, attempted to restart server, and loops are all still open.

![laserweb](https://cloud.githubusercontent.com/assets/920031/15988072/fe21e2a2-3010-11e6-9a14-6bfbb70d03d2.png)

I'd be fine with disabling SVG loading if we can't support it.  At least that sets appropriate user expectations.  Stupid broken Windows copy of Inkscape won't export DXFs for me though.

**Comment by cojarbi** [comment]
Same here. Refreshing after changed setting also does nothing

On Sat, Jun 11, 2016 at 7:14 PM, thantik notifications@github.com wrote:

> No gcode here either @cojarbi https://github.com/cojarbi -
> Additionally, the new option seems to do nothing as far as closing loops.
> 
> [image: laserweb]
> https://cloud.githubusercontent.com/assets/920031/15988072/fe21e2a2-3010-11e6-9a14-6bfbb70d03d2.png
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/openhardwarecoza/LaserWeb3/issues/14#issuecomment-225401715,
> or mute the thread
> https://github.com/notifications/unsubscribe/APrsFkDlbUIVa3j54BecI8r3zqntfxyTks5qK09vgaJpZM4IzkKx
> .

## 

Ariel Yahni

**Comment by ghost** [comment]
thats because I am moving the rapid speeds from settings, remember (;

ctrl+shft+i

**Comment by ghost** [comment]
fixed

**Comment by thantik** [comment]
100% confirmed, tested and working!

**Comment by ghost** [comment]
Great! Last night I twice made capitalization mistakes
